src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs

```diff
@@ -514,9 +514,7 @@ public void SendAsync_ExpectedDiagnosticSynchronousExceptionActivityLogging()
                         {
                             // Forces a synchronous exception for WinHttpHandler
                             handler.UseCookies = true;
-                            handler.CookieContainer = null;
-
-                            Assert.ThrowsAsync<InvalidOperationException>(() => client.GetAsync($"http://{Guid.NewGuid()}.com")).Wait();
+                            Assert.Throws<ArgumentNullException>(() => handler.CookieContainer = null);
```

Comment:
Isn't the test expected to force an exception during SendAsync/GetAsync?

Reply:
You are right, since I changed the place for checking Null for `CookieContainer`, the exception now will not be thrown from SendAsync.

I need to revisit the WinHttpHandler (and CurlHandler) logic to avoid the duplicated check, and potentially force a difference Exception (from SendAsync/GetAsync) for this test.

Reply:
.NET Framework (Desktop) throws the exception in the sync path (i.e. during the property setter) before any SendAsync is done.

>So, the consistent thing to do in the HttpClientHandler layer (above the WinHttpHandler layer on Windows) is probably to be consistent and just throw the exception in the setter. This would keep the same behavior across all platforms for the setter of the HttpClientHandler.CookieContainer property.

Reply:
>Isn't the test expected to force an exception during SendAsync/GetAsync?

I don't think we want that.  The .NET Framework behavior for HttpClientHandler simply checks the property setter for NULL.

We should be careful about how we change this because WinHttpHandler has a different cookie property model (it allows both a CookieContainer and the use of the internal WinHttp cookie storage).  So, I don't think we should make any changes to WinHttpHandler at all.  

This issue which was opened a few years ago was basically about matching exact .NET Framework behaviors for HttpClientHandler.  Given the fact that this issue has been around for a while and we haven't had any customer feedback on it, it might be better to not change this at all and simply close the GitHub issue.

Reply:
> I don't think we want that. The .NET Framework behavior for HttpClientHandler simply checks the property setter for NULL.

This test class only runs for .NET Core. Specifically this block is testing WinHttpHandler's behavior.

> So, I don't think we should make any changes to WinHttpHandler at all.

The change is in HttpClientHandler layer. My original concern was that, the change in HttpClientHandler prevents the Exception from WinHttpHandler SendAsync. This will be a behavior difference if WinHttpHandler is directly used. But for now I don't think the concern is valid, because if customer chooses to use HttpClientHandler, he will expect consistent behavior across Framework and Core.

> Given the fact that this issue has been around for a while and we haven't had any customer feedback on it, it might be better to not change this at all and simply close the GitHub issue.

I'm not 100% convinced. We won't fix because the customer impact is low?

Reply:
Are there any cases in which this will actually cause issues? The default value of `WinHttpHandler.CookieContainer` is null, so we aren't preventing some invalid state from occurring by blocking users from setting it to null explicitly.

Reply:
> Are there any cases in which allowing users to set this value to null will cause issues?
> If we have a reason other than strict compat, that would be more compelling.

It's a compat issue. When set this value to null, Framework will throw at setter, Core will throw at `SendAsync`.
No customer report that this causes issues yet. I don't think there are other reasons than compat.